class.jetpack.php

```diff
@@ -7451,6 +7488,75 @@ public static function is_active_and_not_development_mode() {
 		return true;
 	}
 
+	/**
+	 * Adds custom styles for Jetpack Scan threat indicator.
+	 *
+	 * @since 8.6.0
+	 * @return void
+	 */
+	public function enqueue_toolbar_styles() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		if ( ! self::is_scan_enabled() ) {
+			return;
+		}
+
+		wp_add_inline_style(
+			'admin-bar',
+			'#wp-admin-bar-jp-scan-notice.error { background-color: #D63638; }
+				#wp-admin-bar-jp-scan-notice.error:hover { background-color: #B32D2E; }
+				#wp-admin-bar-jp-scan-notice.error > ab-item { color: #FFF; }'
+		);
+	}
+
+	/**
+	 * Adds Jetpack Scan threat indicator to the WordPress admin bar.
+	 *
+	 * @since 8.6.0
+	 * @param WP_Admin_Bar $wp_admin_bar Reference to admin bar.
+	 * @return void
+	 */
+	public function add_threats_to_toolbar( $wp_admin_bar ) {
+		global $wp_version;
+
+		if ( version_compare( $wp_version, '3.3', '<' ) ) {
+			return;
+		}
+
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		if ( ! self::is_scan_enabled() ) {
+			return;
+		}
+
+		$threat_count = intval( get_transient( 'jetpack_scan_threat_count' ) );
+		if ( $threat_count < 1 ) {
+			return;
+		}
+
+		$threat_count = number_format( $threat_count, 0 );
+		$wp_admin_bar->add_node(
+			array(
+				'id'     => 'jp-scan-notice',
+				'title'  => esc_html(
+					// translators: %s is number of threats found.
+					sprintf( _n( '%s Security Threat', '%s Security Threats', $threat_count, 'jetpack' ), $threat_count )
+				),
+				'parent' => 'top-secondary',
+				'href'   => esc_url( Redirect::get_url( 'jetpack-scan-threat' ) ),
```

Comment:
This also should be updated to use `calypso-scanner`
```suggestion
				'href'   => esc_url( Redirect::get_url( 'calypso-scanner' ) ),
```